File: Services/RandoxITUtility/Domain/Entities/TimeData.cs
using System;
using RandoxITUtility.Domain;

namespace RandoxITUtility.Domain.Entities
{
    public class TimeData
    {
        public Guid Id { get; set; }
        public DateTime WeekCommencing { get; set; } 
        public DateTimeOffset StartTime { get; set; }
        public DateTimeOffset EndTime { get; set; }
        public DateTimeOffset LunchLength { get; set; }
        public string DayOfTheWeek { get; set; }
    }
}

File: WebPortal/src/app/models/Day/TimeData.ts
import { ITimeData } from '../interfaces/Day/ITimeData';
import { Moment } from 'moment';

export class TimeData implements ITimeData {
    id: string;
    dayOfTheWeek: string;
    startTime: Moment;
    endTime: Moment;
    lunchLength: Moment;
    weekCommencing: Moment;

    constructor(id: string, dayOfTheWeek: string, startTime: Moment, endTime: Moment, lunchLength: Moment, weekCommencing: Moment) {
        this.id = id;
        this.dayOfTheWeek = dayOfTheWeek;
        this.startTime = startTime;
        this.endTime = endTime;
        this.lunchLength = lunchLength;
        this.weekCommencing = weekCommencing;
    }

    getTypes() {
        return {}
    }
}
